app/controllers/bike_share_app.rb

```diff
@@ -29,17 +30,19 @@ class BikeShareApp < Sinatra::Base
   end
 
   get "/stations/:id/edit" do
-    ## if you want to have the user select city, installation date, and/or block count with a drop down, add instance variables like the one below.
-    ## @cities = City.all.order(:name)
     @station = Station.find(params[:id])
     erb :"stations/edit"
   end
 
   put "/stations/:id" do
-    ## if you want to have the user select city, installation date, and/or block count with a text field,
-    ##     you must create a new object or find an existing object with the data the user entered and then replace that text with the id in the parameters to create the station
-    ##     params[:city] = City.find_or_create_by(name: params[:city]).id
-    @station = Station.update(params[:id],params[:station])
+    Station.update(name: params[:station][:name],
+     city_id: City.find_or_create_by(params[:city]).id,
+     dock_count_id: DockCount.find_or_create_by(params[:dock_count]).id,
+     installation_date_id: InstallationDate.find_or_create_by(params[:installation_date]).id)
```

Comment:
This is an interesting choice. Can you explain why you'd want to have a date table? Can see some justification for it, but my initial response is similar to my response to DockCount. Dates aren't countable and limited in the way that a station or a trip is, so my initial impulse is to store them as dates.

In either case, likely wouldn't have an `installation_dates` table.